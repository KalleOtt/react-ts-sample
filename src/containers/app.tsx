import * as React from 'react';
import { Component } from 'react';
import { connect } from 'react-redux';
import { Store, Dispatch, bindActionCreators } from 'redux';
import { ApplicationState } from 'models/applicationState';
import { inputChanged } from 'actions/actionCreators';
import { MarkdownText } from 'components/MarkdownText';
import { TextArea } from 'components/TextArea';
import { Card } from 'react-toolbox';
import 'react-toolbox/lib/commons.scss';

interface IAppProps {
    text: string;
    inputChanged: typeof inputChanged;
    dispatch: Dispatch<Store<ApplicationState>>;
}

const mapDispatchToProps = (dispatch: Dispatch<any>) => {
    return { inputChanged: bindActionCreators(inputChanged, dispatch) }
};

const mapStateToProps = (state: ApplicationState) => {
    return {
        text: state.text
    };
};

class AppComponent extends Component<IAppProps, {}> {

    private onValueChanged = (value: string) => {
        const {inputChanged} = this.props;
        if (inputChanged) {
            inputChanged(value);
        }
    }

    render() {
        const {text} = this.props;

        const columnStyle: React.CSSProperties = {
            display: 'flex',
            flexDirection: 'column',
            height: '600px',
            width: '100%'
        };

        const editorWrapperStyle: React.CSSProperties = {
            display: 'flex',
            height: '100%',
            flexGrow: 1,
            justifyContent: 'space-around'
        };

        const paperStyle: React.CSSProperties = {
            height: '100%',
            margin: '10px',
            padding: '8px',
            width: 'calc(50% - 10px)',
            maxWidth: '800px',
            boxSizing: 'border-box',
        };

        return (
            <div style={columnStyle} >
                <div style={editorWrapperStyle}>
                    <Card style={paperStyle}>
                        <TextArea onChange={this.onValueChanged} value={text} />
                    </Card>
                    <Card style={paperStyle}>
                        <MarkdownText text={text} />
                    </Card>
                </div>
            </div>
        );
    }
}

export const App = connect(mapStateToProps, mapDispatchToProps)(AppComponent);