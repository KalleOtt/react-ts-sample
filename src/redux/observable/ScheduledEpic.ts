import { ActionsObservable } from 'redux-observable'
import { MiddlewareAPI } from 'redux'
import { Scheduler } from 'rxjs/Scheduler'
import { Observable } from 'rxjs/Observable'

export type ScheduledEpic<T, S> = (
  action$: ActionsObservable<T>,
  store?: MiddlewareAPI<S>,
  scheduler?: Scheduler
) => Observable<T>
