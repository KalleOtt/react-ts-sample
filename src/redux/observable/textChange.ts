import 'rxjs/add/operator/filter'
import 'rxjs/add/operator/map'
import 'rxjs/add/operator/debounceTime'
import { Action, ActionType, InputChangedAction } from 'actions/actionTypes'
import { textChanged } from 'actions/actionCreators'
import { ApplicationState } from 'models/applicationState'
import { ScheduledEpic } from 'redux/observable/ScheduledEpic'

export const textChangedEpic: ScheduledEpic<Action, ApplicationState> = (
  action$,
  store?,
  scheduler?
) =>
  action$
    .ofType(ActionType.INPUT_CHANGED)
    .map((action: InputChangedAction) => textChanged(action.input))
    .debounceTime(150, scheduler)
