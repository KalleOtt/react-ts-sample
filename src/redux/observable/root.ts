import { combineEpics } from 'redux-observable'
import { textChangedEpic } from './textChange'

export const rootEpic = combineEpics(textChangedEpic)
