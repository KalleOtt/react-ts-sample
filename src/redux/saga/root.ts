import { watchTextChange } from './textChange'
import { SagaIterator } from 'redux-saga'
import { fork } from 'redux-saga/effects'

export const rootSaga: () => SagaIterator = function* rootSaga() {
  yield [fork(watchTextChange)]
}
