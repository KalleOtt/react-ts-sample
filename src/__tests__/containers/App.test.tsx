import * as React from 'react';
import renderer from 'react-test-renderer';
import { App } from 'containers/app';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import { rootEpic } from 'redux/observable/root';

const epicMiddleware = createEpicMiddleware(rootEpic);
const mockStore = configureMockStore([epicMiddleware]);

jest.mock('react-toolbox/lib/input/Input', () => { return { inputFactory: () => 'Input' }; });

describe('App should be rendered', () => {

    const component = renderer.create(
        <Provider store={mockStore()}>
            <App />
        </Provider>
    );
    let tree = component.toJSON();

    test('Snapshot test', () => {
        expect(tree).toMatchSnapshot();
    });
});