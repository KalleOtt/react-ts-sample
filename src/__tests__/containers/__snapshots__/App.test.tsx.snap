// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be rendered Snapshot test 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "600px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexGrow": 1,
        "height": "100%",
        "justifyContent": "space-around",
      }
    }
  >
    <div
      className=""
      data-react-toolbox="card"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "100%",
          "margin": "10px",
          "maxWidth": "800px",
          "padding": "8px",
          "width": "calc(50% - 10px)",
        }
      }
    >
      <Input
        disabled={undefined}
        multiline={true}
        name={undefined}
        onChange={[Function]}
        rows={27}
        style={
          Object {
            "height": "100%",
            "lineHeight": "20px",
            "maxHeight": "100%",
            "overflow": "auto",
          }
        }
        theme={Object {}}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
      data-react-toolbox="card"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "100%",
          "margin": "10px",
          "maxWidth": "800px",
          "padding": "8px",
          "width": "calc(50% - 10px)",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        style={
          Object {
            "color": "black",
          }
        }
      />
    </div>
  </div>
</div>
`;
